### PR TITLE
Add search page and search functionality

### PR DESCRIPTION
**Pull request by berlinum**
+ Add fetch rebrickalble API to get search results.
+ Add "throttling" for search input.
+ Add loading animation. 

__Preview__: https://bricks-pipel-searchsets-jea855.herokuapp.com/

### CODE REVIEW

File: client/src/api/rebrick.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export async function getSets(query) {
+  const url = `https://rebrickable.com/api/v3/lego/sets/?search=${query}`;
```
Comment (lmachens) [inline client/src/api/rebrick.js:2]: You should call this API from your own server to make it possible to hide the authorization key from the client and have more control over the transferred data.
Thus, you should create a new endpoint in your API which forwards the request to the rebrickable API and returns the data back to the client. Maybe convert it if it makes sense (the properties are not in camelCase -> you could map it on server side).

File: client/src/assets/icons/Loading.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+/* eslint-disable react/no-unknown-property */
```
Comment (lmachens) [inline client/src/assets/icons/Loading.js:1]: What is the reason for this eslint-disable?

File: client/src/components/SearchInput.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  font-size: 16px;
+  color: ${colors.textPrimary};
+  &::placeholder {
+    color: ${colors.textInfo};
+  }
+`;
+
+const SearchInput = ({ value, onChange }) => {
+  return (
+    <div>
```
Comment (lmachens) [inline client/src/components/SearchInput.js:34]: This div can be removed

File: client/src/assets/icons/Loading.js (hunk 2 above)
Comment (berlinum) [reply client/src/assets/icons/Loading.js]: Solved. (Changed svg-properties)

File: client/src/api/rebrick.js (hunk 1 above)
Comment (berlinum) [reply client/src/api/rebrick.js]: Done!

File: client/src/components/SearchInput.js (hunk 3 above)
Comment (berlinum) [reply client/src/components/SearchInput.js]: done!

File: lib/routes/search.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+// Load environment variables from .env file
+require('dotenv').config();
+const { getSets } = require('../models/sets');
+const { Router } = require('express');
+
+const router = Router();
+
+router.get('/set', async (request, response) => {
```
[the remark this answers is not in the document]
Comment (berlinum) [reply lib/routes/search.js:8]: Make sense!

File: package.json [hunk 5]
```diff
@@ -51,6 +51,8 @@
     "express": "^4.17.1",
     "jsonwebtoken": "^8.5.1",
     "mongodb": "^3.5.5",
-    "mongoose": "^5.9.7"
+    "mongoose": "^5.9.7",
+    "node-fetch": "^2.6.0",
+    "react-query": "^1.2.9"
```
[the remark this answers is not in the document]
Comment (berlinum) [reply package.json:56]: Right! 🙈

**Pull request merged**